code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/ManualCarouselNotificationBuilderTest.kt

```diff
@@ -0,0 +1,239 @@
+        pushTemplate.intentAction = PushTemplateConstants.IntentActions.MANUAL_CAROUSEL_LEFT_CLICKED
+        val imageUris = listOf("image1", "image2", "image3")
+        val result = getCarouselIndices(pushTemplate, imageUris)
+        assertEquals(Triple(1, 2, 0), result)
+    }
+
+    @Test
+    fun `test getCarouselIndices with filmstrip left click intent action`() {
+        pushTemplate.intentAction = PushTemplateConstants.IntentActions.FILMSTRIP_LEFT_CLICKED
+        val imageUris = listOf("image1", "image2", "image3")
+        val result = getCarouselIndices(pushTemplate, imageUris)
+        assertEquals(Triple(1, 2, 0), result)
+    }
+
+    @Test
+    fun `test getCarouselIndices with no intent action and filmstrip layout`() {
+        pushTemplate.intentAction = PushTemplateConstants.DefaultValues.FILMSTRIP_CAROUSEL_MODE
+        val imageUris = listOf("image1", "image2", "image3")
+        val result = getCarouselIndices(pushTemplate, imageUris)
+        assertEquals(
+            Triple(
+                PushTemplateConstants.DefaultValues.FILMSTRIP_CAROUSEL_CENTER_INDEX - 1,
+                PushTemplateConstants.DefaultValues.FILMSTRIP_CAROUSEL_CENTER_INDEX,
+                PushTemplateConstants.DefaultValues.FILMSTRIP_CAROUSEL_CENTER_INDEX + 1
+            ),
+            result
+        )
+    }
+
+    @Test
+    fun `test populateManualCarouselImages with valid images`() {
+        val packageName = context.packageName
+        val centerIndex = 1
+        every { getCachedImage(any()) } answers { mockkClass(Bitmap::class) }
+        every { expandedLayout.addView(any(), any<RemoteViews>()) } just Runs
+        every { expandedLayout.setDisplayedChild(any(), any()) } just Runs
+
+        populateManualCarouselImages(
+            context,
+            pushTemplate.carouselItems,
+            packageName,
+            centerIndex,
+            pushTemplate,
+            trackerActivityClass,
+            expandedLayout
+        )
+
+        verify { expandedLayout.addView(any(), any<RemoteViews>()) }
+        verify { expandedLayout.setDisplayedChild(any(), centerIndex) }
+    }
+
+    @Test
+    fun `test populateManualCarouselImages with null images`() {
+        val packageName = context.packageName
+        val centerIndex = 1
+        every { getCachedImage(any()) } returns null
+
+        populateManualCarouselImages(
+            context,
+            pushTemplate.carouselItems,
+            packageName,
+            centerIndex,
+            pushTemplate,
+            trackerActivityClass,
+            expandedLayout
+        )
+
+        verify(exactly = 0) { expandedLayout.addView(any(), any<RemoteViews>()) }
+        verify(exactly = 0) { expandedLayout.setDisplayedChild(any(), centerIndex) }
+    }
+
+    @Test
+    fun `test populateFilmstripCarouselImages with valid images`() {
+        val mockBitmap = mockkClass(Bitmap::class)
+        val imageCaptions = listOf("Caption 1", "Caption 2", "Caption 3")
```

Comment:
Can you add a test with more than 3 images in the list to make sure parsing works as expected?